libcnb/src/build.rs

```diff
@@ -72,18 +73,21 @@ impl BuildResultBuilder {
 }
 
 impl BuildResultBuilder {
+    #[must_use]
     pub fn build(self) -> BuildResult {
         BuildResult(InnerBuildResult::Pass {
             launch: self.launch,
             store: self.store,
         })
     }
 
+    #[must_use]
```

Comment:
I don't think `#[must_use]` should be used on this function and the one below given they do mutate the original struct - seems like false positives.

Fixing this lint is definitely one where we need to vet each case and not just add the annotations where clippy suggested without checking :-)